example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,222 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class IosPopoverMenu extends StatelessWidget {
+  const IosPopoverMenu({
+    super.key,
+    this.radius = const Radius.circular(12),
+    this.arrowWidth = 18.0,
+    this.arrowLength = 12.0,
+    required this.arrowDirection,
+    this.backgroundColor = const Color(0xFF474747),
+    this.padding = EdgeInsets.zero,
+    this.arrowFocalPoint,
+    required this.child,
+  });
+
+  /// Radius of the decoration corners.
+  final Radius radius;
+
+  /// Extent of the arrow.
+  final double arrowLength;
+
+  /// Width of the arrow.
+  final double arrowWidth;
+
+  /// Direction where the arrow points to.
+  final ArrowDirection arrowDirection;
+
+  /// Padding around the popover content.
+  final EdgeInsets padding;
+
+  /// Color of the decoration.
+  final Color backgroundColor;
+
+  /// Center point of the arrow.
+  ///
+  /// Defaults to the center of the axis.
+  final double? arrowFocalPoint;
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return DecoratedBox(
+      decoration: ShapeDecoration(
+        color: backgroundColor,
+        shape: _IosMenuShapeBorder(
+          arrowDirection: arrowDirection,
+          arrowFocalPoint: arrowFocalPoint,
+          arrowLength: arrowLength,
+          arrowWidth: arrowWidth,
+          padding: padding,
+          radius: radius,
+        ),
+      ),
+      child: Padding(
+        padding: _getArrowPadding(),
+        child: Padding(
+          padding: padding,
+          child: child,
+        ),
+      ),
+    );
+  }
+
+  /// Returns the padding needed to leave space for the arrow.
+  EdgeInsets _getArrowPadding() {
+    return EdgeInsets.fromLTRB(
+      arrowDirection == ArrowDirection.left ? arrowLength : 0,
+      arrowDirection == ArrowDirection.up ? arrowLength : 0,
+      arrowDirection == ArrowDirection.right ? arrowLength : 0,
+      arrowDirection == ArrowDirection.down ? arrowLength : 0,
+    );
+  }
+}
+
+class _IosMenuShapeBorder extends ShapeBorder {
```

Comment:
Extending `ShapeBorder` is very uncommon. This is definitely a class that should have a Dart Doc, which explains the motivation for creating this custom border, and what the border looks like.

Reply:
Initially, I implemented using a `RenderObject` but then switched to using a `ShapeBorder` because it seemed to be simpler. After the change requested in https://github.com/Flutter-Bounty-Hunters/follow_the_leader/pull/6#discussion_r978260800 I switched back to the `RenderObject`.